File: wp_node.py
#!/usr/bin/env python3
# coding=utf-8

import rospy
from std_msgs.msg import String

def NavResultCallback(msg):
    rospy.logwarn("导航结果 = %s",msg.data)

if __name__=="__main__":
    rospy.init_node("wp_node")

    navi_pub = rospy.Publisher("/waterplus/navi_waypoint",String,queue_size=10)
    res_sub = rospy.Subscriber("/waterplus/navi_result",String,NavResultCallback,queue_size=10)

    rospy.sleep(1)

    navi_msg = String()
    navi_msg.data = "1"
    navi_pub.publish(navi_msg)

    rospy.spin()

File: wp_node.cpp
#include<ros/ros.h>
#include<std_msgs/String.h>

void NavResultCallback(const std_msgs::String &msg)
{
    ROS_WARN("[NavResultCallback] %s",msg.data.c_str());
}

int main(int argc,char** argv)
{
    ros::init(argc,argv,"wp_node");

    ros::NodeHandle n;
    ros::Publisher nav_pub = n.advertise<std_msgs::String>("/waterplus/navi_waypoint",10);
    ros::Subscriber res_sub = n.subscribe("/waterplus/navi_result",10,NavResultCallback);

    sleep(1);
    std_msgs::String nav_msg;
    nav_msg.data = "2";
    nav_pub.publish(nav_msg);

    ros::spin();

    return 0;
}